src/application/ToDoManager.java

```diff
@@ -0,0 +1,96 @@
+package application;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ToDoManager {
+	
+	private List<ToDoItem> toDoList;
+	
+	// Constructor (initialize to-do list)
+    public ToDoManager() {
+        this.toDoList = new ArrayList<>();
+    }
+    
+    public void addTask(ToDoItem task) {
+        toDoList.add(task);
+    }
+    
+    public void deleteTask(ToDoItem task) {
+        toDoList.remove(task);
+    }
+    
+    public void editTask(ToDoItem task, String newTitle, String newDescription, LocalDate newDueDate,
+            			 ToDoItem.Priority newPriority, ToDoItem.Tag newTag) {
+		task.editTask(newTitle, newDescription, newDueDate, newPriority, newTag);
+    }
+    
+    // choose a specific tag
+    public List<ToDoItem> filterTasksByTag(ToDoItem.Tag tag) {
+        List<ToDoItem> filteredTasks = new ArrayList<>();
+        for (int i = 0; i < toDoList.size(); i++) { 
+            ToDoItem task = toDoList.get(i);
+            if (task.getTag() == tag) {
+                filteredTasks.add(task);
+            }
+        }
+        return filteredTasks;
+    }
+    
+    // choose a priority level
+    public List<ToDoItem> filterTasksByPriority(ToDoItem.Priority priority) {
+        List<ToDoItem> filteredTasks = new ArrayList<>();
+        for (int i = 0; i < toDoList.size(); i++) { 
+            ToDoItem task = toDoList.get(i);
+            if (task.getPriority() == priority) {
+                filteredTasks.add(task);
+            }
+        }
+        return filteredTasks;
+    }
+    
+    // choose for a date
+    public List<ToDoItem> getTasksForDay(LocalDate date) {
+        List<ToDoItem> tasksForDay = new ArrayList<>();
+        for (int i = 0; i < toDoList.size(); i++) { 
+            ToDoItem task = toDoList.get(i);
+            if (task.getDueDate().equals(date)) {
+                tasksForDay.add(task);
+            }
+        }
+        return tasksForDay;
+    }
+    
+    public List<ToDoItem> getCompletedTasks() {
+        List<ToDoItem> completedTasks = new ArrayList<>();
+        for (int i = 0; i < toDoList.size(); i++) { 
+            ToDoItem task = toDoList.get(i);
+            if (task.isCompleted()) {
+                completedTasks.add(task);
+            }
+        }
+        return completedTasks;
+    }
+
+    public List<ToDoItem> getPendingTasks() {
```

Comment:
When will we use this method?

Reply:
call this list to print out all the pending tasks on ui?

Reply:
regarding the method of the addTask(), the ToDoItem constructor is used to initialize the properties of a task when creating it. For example, we use new ToDoItem() to create a new task, and then we add it to the ToDoManager using the addTask() method.

In case here is the example:

ToDoItem task = new ToDoItem( // Create a new task using the ToDoItem constructor
    "Buy groceries",             // Title
    "Buy milk, eggs, and bread", // Description
    LocalDate.now(),             // Due date
    ToDoItem.Priority.HIGH,      // Priority
    ToDoItem.Tag.ERRAND          // Tag
);

toDoManager.addTask(task); // Add the task to the ToDoManager

Therefore, addTask() doesn’t create a task by itself. Instead, it takes an already created task (using the ToDoItem constructor) and stores it in the list.

Reply:
About how to handle strings representing dates and times with both year, month, day, hour, and minute, we can use LocalDateTime

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

String customDateTimeString = "16/11/2024 14:30"; // Custom date-time string
DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
LocalDateTime customDateTime = LocalDateTime.parse(customDateTimeString, formatter);

System.out.println(customDateTime); // Output: 2024-11-16T14:30